Integrations/CiscoFirepower/CiscoFirepower.py

```diff
@@ -0,0 +1,1474 @@
+from CommonServerPython import *
+
+''' IMPORTS '''
+from typing import Dict, Tuple, List, Union
+import urllib3
+
+"""Example for Analytics and SIEM integration
+"""
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+'''GLOBALS/PARAMS'''
+
+INTEGRATION_NAME = 'Cisco Fire Power'
+# lowercase with `-` dividers
+INTEGRATION_COMMAND_NAME = 'ciscofp'
+# No dividers
+INTEGRATION_CONTEXT_NAME = 'CiscoFP'
+
+OUTPUT_KEYS_DICTIONARY = {
+    'id': 'ID'
+}
+
+
+class Client(BaseClient):
+    def login(self):
+        """update the X-auth-access-token in the client.
+        """
+        new_headers = self._http_request(
+            'POST',
+            url_suffix='/api/fmc_platform/v1/auth/generatetoken',
+            resp_type='response'
+        ).headers
+        self._headers = {'X-auth-access-token': new_headers.get('X-auth-access-token')}
+        self._base_url += f'/api/fmc_config/v1/domain/{new_headers.get("DOMAIN_UUID")}/'
+        if self._headers['X-auth-access-token'] == '':
+            return_error('No valid access token')
+        return
+
+    def list_zones(self, limit, offset) -> Dict:
+        suffix = f'object/securityzones?expanded=true&limit={limit}&offset={offset}'
+        return self._http_request('GET', suffix)
```

Comment:
It seems to me that `list_zones`, `list_ports` and `list_url_categories` could be in the same function that gets `limit`, `offset` and `object_type`(`securityzones`/`ports`/`urlcategories`) that will change the suffix accordingly.
Plus see line 49.